spacy_layout/adapters/azure_adapter.py

```diff
@@ -0,0 +1,212 @@
+try:
+    from dotenv import load_dotenv
+
+    DOTENV_AVAILABLE = True
+except ImportError:
+    DOTENV_AVAILABLE = False
+
+from .base import BackendAdapter
+
+
+class AzureAdapter(BackendAdapter):
+    """Adapter for Azure AI Document Intelligence."""
+
+    def __init__(self, endpoint: str = None, key: str = None, dotenv_path: str = None):
+        """
+        Initialize the Azure adapter.
+
+        Args:
+            endpoint: Azure Document Intelligence endpoint
+            key: Azure Document Intelligence API key
+            dotenv_path: Path to .env file containing Azure credentials
+        """
+        # Load environment variables from .env file if available
+        if DOTENV_AVAILABLE and dotenv_path is not None:
+            load_dotenv(dotenv_path)
+        elif DOTENV_AVAILABLE:
+            load_dotenv()  # Try to load from default locations
+
+        self.endpoint = endpoint or os.environ.get(
+            "AZURE_DOCUMENT_INTELLIGENCE_ENDPOINT"
+        )
+        self.key = key or os.environ.get("AZURE_DOCUMENT_INTELLIGENCE_KEY")
+
+        if not self.endpoint or not self.key:
+            raise ValueError(
+                "Azure Document Intelligence endpoint and key must be provided or "
+                "set as environment variables (AZURE_DOCUMENT_INTELLIGENCE_ENDPOINT, "
+                "AZURE_DOCUMENT_INTELLIGENCE_KEY) or in a .env file"
+            )
+
+        self.client = DocumentIntelligenceClient(
+            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
+        )
+
+    def _azure_to_docling_document(self, result) -> DoclingDocument:
+        """
+        Convert Azure Document Intelligence result to DoclingDocument.
+
+        Args:
+            result: Azure Document Intelligence analysis result
+
+        Returns:
+            DoclingDocument: Document in Docling format
+        """
+        # Create an empty DoclingDocument
+        doc = DoclingDocument()
+
+        # Process pages
+        for page_idx, page in enumerate(result.pages):
+            # Create a page item (1-indexed in Docling)
+            page_item = PageItem(
+                page_no=page.page_number,
+                size={"width": page.width, "height": page.height},
+                image=None,
+            )
+            doc.pages[page_idx + 1] = page_item
+
+        # Process content
+        item_id = 0
+
+        # Handle paragraphs and text
+        for paragraph in result.paragraphs:
+            page_number = (
+                paragraph.bounding_regions[0].page_number
+                if paragraph.bounding_regions
+                else 1
+            )
+
+            # Create bounding box
+            bbox = None
+            if paragraph.bounding_regions:
+                region = paragraph.bounding_regions[0]
+                # Convert Azure polygon to Docling bounding box
+                x_values = [p.x for p in region.polygon]
+                y_values = [p.y for p in region.polygon]
```

Comment:
I believe `region.polygon` is a list of floats where even-indexed values are x-coordinates and odd-indexed values are y-coordinates.

Reply:
Thanks, you were right. I fixed with:
```python
x_values = region.polygon[::2]
y_values = region.polygon[1::2]
```